app/js/controller/main_controller.js

```diff
@@ -16,10 +20,13 @@ export default class MainController extends Controller {
 	}
 
 	main() {
-		this.view.render();
-		window.addEventListener('hashchanged', this.route.bind(this));
-		this.route();
-		document.body.classList.remove('loading');
+		this.activityHandler.ready.then(route => {
+			window.location.hash = route;
```

Comment:
This part is weird to me. I was intended to use the hash/hashchange to specify top-level view changes, but as you can see we only have 1 view really; the list.

The thing I don't like here is using the hash only for the initial selected tab. Currently, we don't change the hash when selecting different tabs from within the app, and so it seems like we are just using the hash as a way to pass the "selected tab" to the list view when the app loads from an activity. But if we ever add hash routes for new views, this logic may get in the way.

We could change it so hashes trigger tab changes and vice versa, or we could remove the `route` and `hashchanged` listening logic completely from the main controller, and just call `listController.main(initialTab)` directly from the `main()` function in `main_controller`. What do you think?